From 795d69749e0337a5aa7821d40aa53391dfac2ab2 Mon Sep 17 00:00:00 2001
From: Kiran Kelageri <kirankelageri@codeaurora.org>
Date: Mon, 23 Jul 2018 17:17:46 -0700
Subject: [PATCH 02/11] Bluetooth-TWS: Filter TWS devices

Added filter to display TWS devices only
under saved earbuds category.

CRs-Fixed: 2285057
Change-Id: Id254210a4a308360e0f2110d6b81d8a3a780bd38
---
 .../settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java     | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java b/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java
index 55505d078f..8759e6f1d8 100644
--- a/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java
+++ b/src/com/android/settings/bluetooth/SavedBluetoothTwsDeviceUpdater.java
@@ -75,9 +75,10 @@ public class SavedBluetoothTwsDeviceUpdater extends BluetoothDeviceUpdater {
 
     @Override
     public void onConnectionStateChanged(CachedBluetoothDevice cachedDevice, int state) {
+        final BluetoothDevice device = cachedDevice.getDevice();
         if (state == BluetoothAdapter.STATE_CONNECTED) {
             removePreference(cachedDevice);
-        } else if (state == BluetoothAdapter.STATE_DISCONNECTED) {
+        } else if (state == BluetoothAdapter.STATE_DISCONNECTED &&  device.isTwsPlusDevice()) {
             addPreference(cachedDevice);
         }
     }
-- 
2.28.0

